lib/waterline/query/dql/tenant.js

```diff
@@ -0,0 +1,194 @@
+
+/**
+ * Tenant
+ *
+ * @param {Object} criteria
+ * @param {Object} options
+ * @param {Function} callback
+ * @return Deferred object if no callback
+ */
+
+module.exports = function(tenant, cb) {
+  var self = this;
+  // console.log('tenant', arguments);
+  var usage = utils.capitalize(this.identity) + '.tenant(tenantId,callback)';
+
+  if (arguments.length === 0) {
+    //
+    // Error
+    return usageError('Tenant must be passed.', usage, cb);
+  }
+
+  if (typeof tenant === 'function') {
+    // Tenant is required
+    // Error
+    return usageError('Tenant must be passed.', usage, cb);
+  }
+
+
+  // Change connection settings based on tenant
+  // console.log('collections', this.waterline.collections);
+  // console.log('adapterDictionary', this.waterline, this.adapterDictionary);
+
+  var baseCollectionIdentity = self.baseIdentity || self.identity;
+  var connectionName = this.adapterDictionary['identity'];
+  var baseConnectionName = connectionName;
+  if (baseCollectionIdentity != self.identity) {
+      // Is not the base Collection
+      // Get base collection
+      var baseCollection = self.waterline.collections[baseCollectionIdentity];
+    //   console.log('baseCollection', baseCollectionIdentity);
+      baseConnectionName = baseCollection.adapterDictionary['identity'];
+    //   console.log('baseConnectionName', baseConnectionName);
+  }
+  var connection = this.connections[connectionName];
+  // Change for Multi-Tenancy support
+  // console.log('connection', connection);
+  var config = connection.config;
+  // console.log('config', config);
+
+  if (config.isMultiTenant && typeof config.getTenant === "function") {
+
+    var configForTenant = config.configForTenant;
+
+    // Return Deferred or pass to adapter
+    if (typeof cb === 'function') {
+
+      var configDup = _.cloneDeep(config);
+      configDup.tenant = tenant; // Set Tenant
+      configForTenant.call(self, tenant, configDup, function(err, config) {
+        // console.log('configForTenant', err, config);
+        if (err) return cb(err);
+
+        // Tenant's config namespace
+        var tenantSuffix = "<tenant:" + tenant + ">";
+        var newConnectionIdentity = (baseConnectionName + tenantSuffix).toLowerCase();
+
+        function createTenantCollection() {
+
+            var newCollectionIdentity = (baseCollectionIdentity + tenantSuffix).toLowerCase();
+
+            // Take new config and register it's connection
+            // var cols = self.waterline._collections;
+            // console.log('collections', cols);
+            // var col = _.find(cols, function(col) {
+            //     return true;
+            // });
+            // var col = self.collection;
+            // console.log(col);
+            // var colDupModel = _.cloneDeep(self);
+            // var Collection = require('../../collection');
+            //
+            // console.log('connections', self.connections);
+            var colDupModel = self.constructor.extend({
```

Comment:
The above is fixed by https://github.com/Glavin001/waterline/commit/1038dabeec7960638942e798aea6886e5ea2d874

I have tested and confirmed that this works for Sails-mongo now. I would presume it to work on all adapters now!